### PR TITLE
[exporter/logging] Build exporter's logger from service's logger

### LINKED ISSUE #5652: [exporter/logging] Logging exporter logger does not honor telemetry logs configuration
**Describe the bug**

The logging exporter does not honor telemetry logs configuration, since a new logger is created at startup which is completely independent from the Collector's logger.

https://github.com/open-telemetry/opentelemetry-collector/blob/feab9491538a882737a5bceb8757b4458a86edd3/exporter/loggingexporter/factory.go#L87

I am not sure if this was intentional, but it is confusing to me :)

**Steps to reproduce**

Run the Collector with the following configuration:

```
receivers:
  hostmetrics:
    scrapers:
      load:
    collection_interval: 3s

exporters:
  logging:
    loglevel: debug

service:
  telemetry:
    logs:
      output_paths: [example.log]
  pipelines:
    metrics:
      receivers: [hostmetrics]
      exporters: [logging]
```

**What did you expect to see?**

I expected logging exporter logs to be logged to the `example.log` file, as this is where the rest of components will log.

**What did you see instead?**

All logs but the logging exporter's are logged to `example.log`. The logging exporter logs are logged to stderr.

**What version did you use?**
Version: v0.55.0

**Additional context**

This also makes this exporter less useful on context where the Collector is used as a library, since you effectively can't control where the logging exporter outputs.
 
This can be easily fixed if we take the core from the main logger and build a new core with sampling options to it, using [`zapcore.NewSamplingWithOptions`](https://pkg.go.dev/go.uber.org/zap/zapcore#NewSamplerWithOptions). The level can also be managed with `zapcore.NewIncreaseLevelCore`, although in this case we lose the possibility of having a lower log level on the logging exporter than in the global one (IMO that's fine so long as we have a descriptive error).

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:**

Build the logger on the logging exporter's logger from the one on `TelemetrySettings`, so that the logger honors the configuration settings on `telemetry::logs` and the configuration options on `CollectorSettings.LoggingOptions`.

Depends on #5678

**Link to tracking Issue:** Fixes #5652

**Documentation:** Documented the feature gate.

### CODE REVIEW

**Comment by mx-psi** [comment]
Some alternatives I considered:

- Log everything as 'info', and use the `loglevel` field to consider what to log.
- Ignore `loglevel` when lower than `telemetry::logs::level`

I think ultimately the right choice is to get rid of `loglevel` and just rely on `telemetry::logs::level` (but I will leave that for another PR :) )

**Review by tigrannajaryan — CHANGES_REQUESTED** [review]
Arguably, the current behavior (not this PR's behavior) is desirable. When I am troubleshooting I often add a `logging` exporter and set it to `debug` level. I don't also want to enable `debug` level for everything else, it will be just too much noise in the logs.

If we feel there is inconsistency because `logging` exporter's `debug` level does not honour the main level setting, I suggest we do something else instead: the "loglevel" setting of the `logging` exporter should control whether the detailed output is generated, _however_ that detailed output should be generated using `info` level, not `debug` level. So, essentially "loglevel" setting will control the behavior of the `logging` exporter, not the behavior of the `logging` exporter's logger. All output from `logging` exporter will happen at the `info` level. Once that change is made you can then pass the service's logger to the `logging` exporter to use.

**Comment by mx-psi** [comment]
Right, this was the first alternative mentioned in https://github.com/open-telemetry/opentelemetry-collector/pull/5677#issuecomment-1181917506. It feels a bit weird to me, but I understand your point about wanting to filter the logs.

PTAL at #5678 which implements the suggested change. I will update this PR once that one is merged, marking as draft until then

File: exporter/loggingexporter/factory.go [hunk 1]
```diff
@@ -16,21 +16,39 @@ package loggingexporter // import "go.opentelemetry.io/collector/exporter/loggin
...
 )
 
 const (
 	// The value of "type" key in configuration.
 	typeStr                   = "logging"
 	defaultSamplingInitial    = 2
 	defaultSamplingThereafter = 500
+
+	// CreateLoggerFromCoreFeatureGate is a feature gate that, when enabled will create the logger from the service logger's core.
+	CreateLoggerFromCoreFeatureGate = "exporter.logging.CreateLoggerFromCore"
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply exporter/loggingexporter/factory.go]: There are now no breaking changes

**Comment by mx-psi** [comment]
Rewritten to not have breaking changes; depends on #5678

**Comment by mx-psi** [comment]
@tigrannajaryan this is ready for another look, I updated the PR description; this no longer introduces any breaking changes

File: exporter/loggingexporter/factory.go [hunk 2]
```diff
@@ -53,50 +54,29 @@ func createDefaultConfig() config.Exporter {
...
-	// We take development config as the base since it matches the purpose
-	// of logging exporter being used for debugging reasons (so e.g. console encoder)
-	conf := zap.NewDevelopmentConfig()
-	conf.Level = zap.NewAtomicLevelAt(cfg.LogLevel)
-	conf.Sampling = &zap.SamplingConfig{
-		Initial:    cfg.SamplingInitial,
-		Thereafter: cfg.SamplingThereafter,
-	}
+func createLogger(cfg *Config, logger *zap.Logger) *zap.Logger {
+	core := zapcore.NewSamplerWithOptions(
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply exporter/loggingexporter/factory.go:74]: This changes the encoding to whatever is set on `service::telemetry::logs::encoding`, but that's intended. Why would you want to have a different encoding for the logging exporter?  If I pass options to `CollectorSettings::LoggingOptions` or modify the `service::telemetry::logs` section, I expect this exporter to follow these settings, just like any other component would. I think the example on this configuration file is specially telling: https://github.com/open-telemetry/opentelemetry-collector/issues/5652#issue-1298642171; it's rather unusual to have an exporter report to `stderr` when the rest of the components are reporting to a file.

It's not possible to change the encoder while preserving other options on zap, so we have to choose between preserving the encoding (even when the user is setting the encoding to something else) or honoring the other options.

**Review by tigrannajaryan — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**